.pre-commit-config.yaml

```diff
@@ -43,12 +43,11 @@ repos:
     rev: 5.11.4
     hooks:
       - id: isort
-  - repo: https://github.com/PyCQA/flake8
-    rev: "6.0.0"
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.0.226
     hooks:
-      - id: flake8
-        additional_dependencies: ["flake8-bugbear==22.7.1"]
-        language_version: python3.9
+      - id: ruff
+        args: [--force-exclude, --format, grouped, --show-source]
```

Comment:
Do we need `show-source`? IMO, the default looks just like a normal pre-commit check, and the paths are in the correct format for quick linking to source (in iterm, vscode, etc).

Reply:
I also think we can use `--fix`; depending on what checks end up getting added, we can always disable a fix for a check if we don't like it.

Reply:
> Do we need show-source?

eh, I just like having the error highlighted. I'll take it out.

> I also think we can use --fix; depending on what checks end up getting added, we can always disable a fix for a check if we don't like it.

How does that work in CI with pre-commit?  Wil pre-commit automatically commit fixes?

Reply:
> How does that work in CI with pre-commit? Will pre-commit automatically commit fixes?

Yes. If you need I could look for an example commit.

Reply:
Thanks, I'll add `--fix` then.

Reply:
In the next version of the pre-commit action (AKA next version of ruff), you can drop the `--force-exclude` too. https://github.com/charliermarsh/ruff-pre-commit/pull/20

Reply:
Added `--fix` and dropped `--force-exclude`.

Reply:
Wow, that was fast. 👍

Reply:
If you wanted an example: https://github.com/pypa/cibuildwheel/pull/1405/commits/6857bd5a3bcb019a4b0a2881bd08b0800d81c0de